x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -90,42 +88,44 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
             return;
         }
 
-        if (XPackSettings.SECURITY_ENABLED.get(settings)) {
-            if (extractClientCertificate) {
-                HttpChannel httpChannel = request.getHttpChannel();
-                SSLEngineUtils.extractClientCertificates(logger, threadContext, httpChannel);
-            }
+        if (enabled == false) {
+            doHandleRequest(request, channel, client);
+            return;
+        }
```

Comment:
👍 
It may not be a conventional Java style, but I personally like this style of "if + return" for handling exceptional cases earlier on.